pkg/api/v1/types.go

```diff
@@ -2110,6 +2110,8 @@ type PodSpec struct {
 	// If specified, the pod's scheduling constraints
 	// +optional
 	Affinity *Affinity `json:"affinity,omitempty" protobuf:"bytes,18,opt,name=affinity"`
+	// SchedulingMismatchedPredicateResults Summary of false podFitsOnNode predicates for investigating unscheduled pods.
```

Comment:
Needs a better description here.  As a naive reader I wouldn't know what this does.

Also, don't use maps in public API results as per API conventions, you should be returning a `[]PredicateResult` that has the necessary fields and use strategic merge patch.